main.js

```diff
@@ -1,3 +1,113 @@
+ * @param {Element=} opts.target - element to be observed
+ *
+ * @returns {Promise<Element>} the value passed will be a single element matching the selector, or whatever the function returned
+ */
+function elementReady(getter, opts = {}) {
+    return new Promise((resolve, reject) => {
+        opts = Object.assign({
+            timeout: 0,
+            target: document.documentElement
+        }, opts);
+        const returnMultipleElements = getter instanceof Array && getter.length === 1;
+        let _timeout;
+        const _getter = typeof getter === 'function' ?
+            (mutationRecords) => {
+                try {
+                    return getter(mutationRecords);
+                } catch (e) {
+                    return false;
+                }
+            } :
+            () => returnMultipleElements ? document.querySelectorAll(getter[0]) : document.querySelector(getter)
+        ;
+        const computeResolveValue = function (mutationRecords) {
+            // see if it already exists
+            const ret = _getter(mutationRecords || {});
+            if (ret && (!returnMultipleElements || ret.length)) {
+                resolve(ret);
+                clearTimeout(_timeout);
+
+                return true;
+            }
+        };
+
+        if (computeResolveValue(_getter())) {
+            return;
+        }
+
+        if (opts.timeout)
+            _timeout = setTimeout(() => {
+                const error = new Error(`elementReady(${getter}) timed out at ${opts.timeout}ms`);
+                reject(error);
+            }, opts.timeout);
+
+
+        new MutationObserver((mutationRecords, observer) => {
+            const completed = computeResolveValue(_getter(mutationRecords));
+            if (completed) {
+                observer.disconnect();
+            }
+        }).observe(opts.target, {
+            childList: true,
+            subtree: true
+        });
+    });
+}
+
+
+function getMenuItems() {
+    const menuItems = document.querySelectorAll('#top_nav .hdtb-mitem a, a[jsname="ONH4Gc"]');
+    let menuItemsObj = {};
+
+    for (const menuItem of menuItems) {
+        menuItemsObj[menuItem.innerText] = menuItem;
+    }
+
+    return menuItemsObj;
+}
+
+window.addEventListener("keydown", function handleNavigation (event) {
+    if (window.document.activeElement !== window.document.body) {
+        return;
+    }
+
+    const keyCode = event.code;
+    if (keyCode.indexOf("Digit") === 0) {
+        return;
+    }
+
+    event.stopPropagation();
+
+    const keyname2category = {
+        "KeyA": "All",
+        "KeyI": "Images",
+        "KeyM": "Maps",
+        "KeyN": "News",
+        "KeyS": "Shopping",
+        "KeyV": "Videos",
+        "KeyB": "Books",
+        "KeyC": "Finance",
+        "KeyF": "Flights",
+        "KeyW": "Weather",
+        "KeyY": "YouTube",
+    };
+    console.log('keyCode', keyCode);
```

Comment:
Would you mind removing any console.log statements?

Reply:
Yes absolutely, will get on it when I get back on my PC

Reply:
ok done, I added everything